source/server/listener_manager_impl.h

```diff
@@ -260,21 +260,38 @@ class ListenerImpl : public Network::ListenerConfig,
   // Network::DrainDecision
   bool drainClose() const override;
 
+  // Network::FilterChainManager
+  const Network::FilterChainSharedPtr
+  findFilterChain(const std::string& transport_socket_name,
+                  const std::string& server_name) const override;
+
   // Network::FilterChainFactory
-  bool createNetworkFilterChain(Network::Connection& connection) override;
+  bool
+  createNetworkFilterChain(Network::Connection& connection,
+                           const std::vector<Network::NetworkFilterFactoryCb>& factories) override;
   bool createListenerFilterChain(Network::ListenerFilterManager& manager) override;
 
   // Configuration::TransportSocketFactoryContext
   Ssl::ContextManager& sslContextManager() override { return parent_.server_.sslContextManager(); }
   Stats::Scope& statsScope() const override { return *listener_scope_; }
 
 private:
+  void addFilterChain(const std::string& transport_socket_name,
+                      const std::vector<std::string>& server_names,
+                      Network::TransportSocketFactoryPtr&& transport_socket_factory,
+                      std::vector<Network::NetworkFilterFactoryCb> filters_factory);
+  static bool isWildcardServerName(const std::string& name);
+
+  std::unordered_map<std::string, std::unordered_map<std::string, Network::FilterChainSharedPtr>>
```

Comment:
Please comment on what the two strings are in the map keys. I think it's transport-socket-type and SNI value.

Reply:
Done.

Also, as I was writing comments and optimizing the code, I've started questioning my previous choice of having 2 separate maps (one for exact server names and one for the wildcard domains), so I merged them together.